### ISSUE TITLE
new classads for MONIT in task dagman and wrapper jobs are wrongly quoted

### ISSUE THREAD

**Issue by belforte** [dmwm/CRABServer#5913]
e.g.

```
[root@vocms0121 task_process]# condor_q -con 'jobuniverse==5&&jobstatus==2' -af CMS_WMTool|uniq
USER
[root@vocms0121 task_process]# condor_q -con 'jobuniverse==7&&jobstatus==2' -af CMS_WMTool|uniq
undefined
"USER"

```
etc.
affected ads are (from Kibana):
```

t data.CMS_TaskType     "cmsRun"
t data.CMS_Type     "Analysis"
t data.CMS_WMTool     "USER"
t data.Type     "analysis" 
```

in a different view:
```
root@vocms0121 task_process]# condor_q 40541692.0 -l|grep CMS_WMTool  
CMS_WMTool = "\"USER\""
[root@vocms0121 task_process]# 

```

**Comment by belforte** [comment]
something must have gone wrong with https://github.com/dmwm/CRABServer/issues/5890

**Comment by belforte** [comment]
offending lines should be these
https://github.com/dmwm/CRABServer/blob/48b3700c0f0feaf273f1ae766e53da61edcf0b6c/src/python/TaskWorker/Actions/DagmanSubmitter.py#L452-L457
But they appear OK. Yet it is curious that  `dagAd["CMS_SubmissionTool"] = "CRAB"` works.

In the SPOOL_DIR the problem appears e.g. in the file `subdaq.ad`:
```
[root@vocms0121 cluster40459263.proc0.subproc0]# cat subdag.ad |grep CMS_
+CMS_SubmissionTool = "CRAB"
+CMS_WMTool = "\"USER\""
+CMS_TaskType = "\"cmsRun\""
+CMS_Type = "\"Analysis\""
+CMS_JobType = "Analysis"
[root@vocms0121 cluster40459263.proc0.subproc0]# 
```

**Comment by belforte** [comment]
with ref to lines like
https://github.com/dmwm/CRABServer/blob/48b3700c0f0feaf273f1ae766e53da61edcf0b6c/src/python/TaskWorker/Actions/DagmanSubmitter.py#L455
the problem is that `info['..']` already contains a quoted string instead of a string for those attributes
```
(Pdb) info['cms_wmtool']
'"USER"'
```
so that we end with different syntax for the various Ads :
```
(Pdb) dagAd["CMS_WMTool"]
'"USER"'
(Pdb) dagAd["CMS_SubmissionTool"]         
'CRAB'
(Pdb) 
```
yet most strings in `info` dictionary are quoted in that way.

Indeed those attributes are passed from DagmanCreator to the schedd via the file `Job.submit` where they are all quoted in the same way:
```
...
+TaskType = "Job"
+CRAB_SubmitterIpAddr = "188.184.31.49"
+CRAB_TaskLifetimeDays = 30
+CRAB_TaskEndTime = 1565253027
+CRAB_SplitAlgo =  "EventBased"
+CRAB_AlgoArgs = "{\"splitOnRun\": false, \"events_per_job\": {\"halt_job_on_file_boundaries\": false, \"events_per_lumi\": 100, \"algorithm\": \"EventBased\", \"applyLumiCorrection\": true, \"runs\": [], \"lumis\": [], \"splitOnRun\": false, \"events_per_job\": 200}, \"halt_job_on_file_boundaries\": false}"
+CMS_WMTool = "USER"
+CMS_TaskType = "cmsRun"
+CMS_SubmissionTool = "CRAB"
+CMS_Type = "Analysis"
+DESIRED_Archs = "X86_64"
...
```
so the syntax in `info` appears OK. The problems seems in turning them into Ads 
https://github.com/dmwm/CRABServer/blob/48b3700c0f0feaf273f1ae766e53da61edcf0b6c/src/python/TaskWorker/Actions/DagmanSubmitter.py#L485-L486

we should not quote again strings which already contain the double quote.

**Comment by belforte** [comment]
that commit was not enough, I still get:
```
belforte@vocms0144/SPOOL_DIR> condor_q -con 'crab_reqname=="190710_232148:belforte_crab_20190711_012129"' -af:h jobuniverse CMS_WMtool CMS_tasktype cms_type cms_submissiontool 
jobuniverse CMS_WMtool CMS_tasktype cms_type   cms_submissiontool
7           "USER"     "cmsRun"     "Analysis" CRAB              
12          "USER"     "cmsRun"     "Analysis" CRAB              
5           USER       cmsRun       Analysis   CRAB              
5           USER       cmsRun       Analysis   CRAB
```
there are fishy lines in `adjust_out.txt`:
```
belforte@vocms0144/SPOOL_DIR> cat adjust_out.txt |grep CMS
        CMS_WMTool = "\"USER\""; 
        DESIRED_CMSDataset = "/GenericTTbar/HC-CMSSW_9_2_6_91X_mcRun1_realistic_v2-v2/AODSIM"; 
        CMS_SubmissionTool = "CRAB"; 
        CMS_JobType = "Analysis"; 
        CMS_Type = "\"Analysis\""; 
        CRAB_JobSW = "CMSSW_10_2_6"; 
        CMS_TaskType = "\"cmsRun\""; 
        CMSGroups = "/cms,/cms/TEAM,T3_US_FNALLPC,/cms/integration,T3_CH_CERN_CAF,T2_CH_CERN,/cms/itcms,/cms/ALARM"; 
belforte@vocms
```0144/SPOOL_DIR>

**Comment by belforte** [comment]
indeed the problem comes from `dagAd` i.e. the fact that  `info['cms_wmtool']` contains `'"USER"'` instead of `'USER'`
since the submission is done with
https://github.com/dmwm/CRABServer/blob/48b3700c0f0feaf273f1ae766e53da61edcf0b6c/src/python/TaskWorker/Actions/DagmanSubmitter.py#L506

**Comment by belforte** [comment]
the bad formatting comes from 
https://github.com/dmwm/CRABServer/blob/9e8efac5974ae2fd4086e3218b83ea4dafbbf87f/src/python/TaskWorker/Actions/DagmanCreator.py#L251
which has no useful comment as to what its exact purpose is, nut looks like the new ads for MONIT have been added to a piece of code which converts  to string via `json.dumps` which indeed transform everyting to double quoted strings `'"..."'`

```
(Pdb) 
> /data/user/CRABServer/src/python/TaskWorker/Actions/DagmanCreator.py(531)makeJobSubmit()
-> info = transform_strings(info)
(Pdb) info['cms_wmtool']
'USER'
(Pdb) n
> /data/user/CRABServer/src/python/TaskWorker/Actions/DagmanCreator.py(532)makeJobSubmit()
-> info['faillimit'] = task['tm_fail_limit']
(Pdb) info['cms_wmtool']
'"USER"'
(Pdb) 
```

**Comment by belforte** [comment]
that was a red-herring. DagmanCreator is fine since it prepares double quoted strings to use in submit files. The issue is in going from `info[]` to `dagAd[]` in DagmanSubmitter. Other fields like `info['userhn']` are coverted properly inside `dagAd` while the new attributes for MONIT are not.

**Comment by belforte** [comment]
somehow the `info` dictionary passed as argument to DagmanSubmitter has two kind of entries, those quoted once (like `'user_proxy': '/data/certs/creds/cfa0cb32f2043a8042e1290debf652b8fd62334e'`) and those quoted twice (like `'userhn': '"belforte"'` or `'cms_wmtool': '"USER"'`)
When used at the right side of `dagAd["..."]=`, the former is OK, while the latter results in `"\"USER\""`.

So two use cases are mixed inside `info[]` , things to be put in submit JDL file whiich need the `"` and things which will only end in the dagAd for the submission where the string should not jave double quotes around it. There's no general way to separate them but to know what each item is and handle it properly.
Bottom line: I need to go for ad-hoc hacking.

**Comment by belforte** [comment]
I think I found it. The new ads must be listed in `SUBMIT_INFO`
https://github.com/dmwm/CRABServer/blob/9e8efac5974ae2fd4086e3218b83ea4dafbbf87f/src/python/TaskWorker/Actions/DagmanSubmitter.py#L40-L42
so that they are converted to classAds via `addCRABInfoToClassAd(ad, info)` which handles correctly the multiple quoting.
https://github.com/dmwm/CRABServer/blob/9e8efac5974ae2fd4086e3218b83ea4dafbbf87f/src/python/TaskWorker/Actions/DagmanSubmitter.py#L95-L102

**Comment by belforte** [comment]
that does it, and I do not need the `strip` added in previous commit.
Will make a new PR and test it more extensively.
